specification/sections/02-providers.md

```diff
@@ -111,59 +111,22 @@ ResolutionDetails<number> resolveNumberValue(string flagKey, number defaultValue
 ResolutionDetails<MyStruct> resolveStructureValue(string flagKey, MyStruct defaultValue, context: EvaluationContext, options: FlagEvaluationOptions);
 ```
 
-#### Context Transformation
+#### Provider hooks
 
-Feature flag management systems often define structures representing arbitrary contextual data pertaining to the runtime, user, or application. The context transformer defines a simple interface to transform the OpenFeature `evaluation context` to such a structure, mapping values appropriately.
-
-See [evaluation context](./03-evaluation-context.md).
+Feature flag management systems often need to transform the context structures the user provides or general address lifecycle concerns. A `provider hook` provides a mechanism for providers to register [`Hook`s](./04-hooks.md) to tap into that lifecycle.
 
 ##### Requirement 2.10
 
-> The provider interface **MAY** define a `context transformer` method or function, which can be optionally implemented in order to transform the `evaluation context` prior to flag value resolution.
-
-The OpenFeature `client` might apply the transformer function before passing the returned value (the `transformed context`) to the provider resolution methods, thus allowing the provider implementation to avoid implementing and calling such transformation logic repeatedly in flag value resolution methods.
+> The provider interface **MAY** define a `provider hook` method or function, which can be optionally implemented in order to add `hook` instances to the client.
```

Comment:
I'm not sure if this needs to be a method or function. For example, in TypeScript it could just be a property on the class.